bluesky/callbacks/best_effort.py

```diff
@@ -161,7 +173,7 @@ def descriptor(self, doc):
         if stream_name == self.dim_stream:
             # Ensure that no independent variables ('dimensions') are
             # duplicated here.
-            columns = [c for c in columns if c not in self.dim_fields]
+            columns = [c for c in columns if c not in self.independent_vars]
```

Comment:
This will also remove the extra dimensions from the table which I suspect they do not want.

Reply:
It only will if `dim_fields` contains *all* the motors. In some cases it does not. Currently, `dim_fields` is interpreted as the number of independent axes (so one means 1d plot, two means 2d plot)
in the case of inner_product_scan  something like dimensions = (['motor1', 'motor2'], 'primary') was transformed into dimensions= (['motor1'], 'primary') (where it's just assumed we plot versus first motor)
 (done here: https://github.com/NSLS-II/bluesky/pull/963/files#diff-3b88b6e90432890fa4a5951ebfc3b6d0L115)